comp/core/config/config.go

```diff
@@ -75,6 +75,15 @@ func newMock(deps mockDependencies, t testing.TB) Component {
 	// call InitConfig to set defaults.
 	config.InitConfig(config.Datadog)
 
+	if deps.Params.ConfigYaml != "" {
+		config.Datadog.SetConfigType("yaml")
+		err := config.Datadog.ReadConfig(strings.NewReader(deps.Params.ConfigYaml))
+		if err != nil {
+			// The YAML was invalid, fail initialization of the mock config.
+			return nil
```

Comment:
Is there a reason to not update the signature of `newMock` and return `(Component, error)`?

Reply:
Oh yeah - that's better. Fixed